compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -1752,6 +1762,226 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         err.emit()
     }
 
+    fn extract_derive_bounds(
```

Comment:
The new suggestion logic starting here isn't... the cleanest and would love it if anyone comes up with a neater alternative, but this is what I got to.